extensions/interactions/TextInput/TextInputPredictionService.js

```diff
@@ -38,9 +38,12 @@ oppia.factory('TextInputPredictionService', [
         if (textInputTokens !== null) {
           var textVector = CountVectorizerService.vectorize(
             textInputTokens, cvVocabulary);
-          prediction = SVMPredictionService.predict(svmData, textVector);
+          predictionResult = SVMPredictionService.predict(svmData, textVector);
         }
-        return prediction;
+        if (predictionResult.predictionConfidence > 0.7) {
```

Comment:
Ditto -- perhaps explain where 0.5 comes from and under which circumstances we might want to modify it; consider making it into a named constant.

Reply:
So there is no rationale behind these choices. These choices passed the frontend tests for correctness and accuracy. However, I am thinking of moving this check to answer classification service and not do it in individual classifiers. That would require me to implement probability calculation for KNN classifier used in code classifier. Once I add that function, I will move this check to answer classification service.

We can modify it and the choice for this should be based on F1 score of classifier (recall vs accuracy), but given that F1 score is different for different classifiers, there is no universal choice that would fit because in our case every different state of exploration is a different classifier. Hence, right not we should choose a number that is acceptable in standard practice.

Reply:
I don't have a strong opinion on where the check should go. But if you move it to the answer classification service wouldn't you still need to branch on the interaction type to know whether to compare against 0.5 or 0.7? If you do need to branch on the interaction type then maybe keeping it in an interaction-specific class like this one might be better.

The other notes in your comment provide great context; we should add this as comments to the relevant files in the code. I think the second paragraph which explains the "why" is quite important.

Reply:
> I don't have a strong opinion on where the check should go. But if you move it to the answer classification service wouldn't you still need to branch on the interaction type to know whether to compare against 0.5 or 0.7?

That is the design decision we have to make. Should we use individual thresholds or a single threshold for all interactions? I think we should go with the individual approach as I have implemented as of now. Although, I don't have much idea on what these limits should be and I think we should keep them both to be 0.7. What do you think?

Reply:
I'm fine with the individual approach that you have implemented, too. I'm also fine with whatever constant we use as long as the rationale (or lack thereof) is clearly documented, so that future developers know the reason behind the constant used and don't treat it as "sacred".